src/makeInfinityAdder.js

```diff
@@ -6,7 +6,25 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let collection = 0;
+
+  const adder = (...nums) => {
+    if (!nums.length) {
+      const sum = collection;
+
+      collection = 0;
+
+      return sum;
+    }
+
+    // If the function takes more than one parameter
+    // sum it with reduce
+    collection += nums.reduce((acc, n) => acc + n, 0);
```

Comment:
```suggestion
  let sum = 0;

  const adder = (...nums) => {
    if (!nums.length) {
      const result = sum;

      sum = 0;

      return result;
    }

    // If the function takes more than one parameter
    // sum it with reduce
    sum += nums.reduce((acc, n) => acc + n, 0);
```
Consider better variable names.